letsencrypt/display/util.py

```diff
@@ -403,6 +411,117 @@ def _get_valid_int_ans(self, max_):
 
         return OK, selection
 
+class NoninteractiveDisplay(object):
+    """File-based display."""
+
+    zope.interface.implements(interfaces.IDisplay)
+
+    def __init__(self, outfile):
+        super(NoninteractiveDisplay, self).__init__()
+        self.outfile = outfile
+
+    def _interaction_fail(self, message, cli_flag, extra=""):
+        "Error out in case of an attempt to interact in noninteractive mode"
+        msg = "Missing command line flag or config entry for this setting:\n"
+        msg += message
+        if extra:
+            msg += "\n" + extra
+        if cli_flag:
+            msg += "\n\n(You can set this with the {0} flag)".format(cli_flag)
+        raise errors.MissingCommandlineFlag, msg
+
+    def notification(self, message, height=10, pause=False):
+        # pylint: disable=unused-argument
+        """Displays a notification without waiting for user acceptance.
+
+        :param str message: Message to display to stdout
+        :param int height: No effect for NoninteractiveDisplay
+        :param bool pause: The NoninteractiveDisplay waits for no keyboard
+
+        """
+        side_frame = "-" * 79
+        message = _wrap_lines(message)
+        self.outfile.write(
+            "{line}{frame}{line}{msg}{line}{frame}{line}".format(
+                line=os.linesep, frame=side_frame, msg=message))
+
+    def menu(self, message, choices, default=None, cli_flag=None, **kwargs):
```

Comment:
I guess my view is that plugins **should** use some version of `**unused` when implementing these interfaces, so that they are extensible in a backwards-compatible way.  That's more true of third party plugins that it is of our own internal implementations, but since people will tend to read our implementations, I'm inclined to lead the way.